src/loader.ts

```diff
@@ -197,7 +197,7 @@ export class Loader {
 
     protected setupWorker(): void {
         // @ts-expect-error The path to the worker source is only during build.
-        this._worker = new Worker(MAIN_WORKER_SOURCE);
+        this._worker = new Worker(new URL(__MAIN_WORKER_SOURCE, import.meta.url), { type: 'module' });
```

Comment:
It's not possible to put the url in a variable, because Webpack only recognizes the pattern for a worker with the url inside the worker constructor. I'll use newlines instead.